Src/FluentAssertions/Equivalency/DataColumnEquivalencyStep.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+
+using FluentAssertions.Data;
+using FluentAssertions.Execution;
+
+namespace FluentAssertions.Equivalency
+{
+    public class DataColumnEquivalencyStep : IEquivalencyStep
+    {
+        /// <summary>
+        /// Gets a value indicating whether this step can handle the current subject and/or expectation.
+        /// </summary>
+        public bool CanHandle(IEquivalencyValidationContext context, IEquivalencyAssertionOptions config)
+        {
+            return typeof(DataColumn).IsAssignableFrom(config.GetExpectationType(context));
+        }
+
+        /// <summary>
+        /// Applies a step as part of the task to compare two objects for structural equality.
+        /// </summary>
+        /// <value>
+        /// Should return <c>true</c> if the subject matches the expectation or if no additional assertions
+        /// have to be executed. Should return <c>false</c> otherwise.
+        /// </value>
+        /// <remarks>
+        /// May throw when preconditions are not met or if it detects mismatching data.
+        /// </remarks>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Style", "IDE0019:Use pattern matching", Justification = "The code is easier to read without it.")]
+        public bool Handle(IEquivalencyValidationContext context, IEquivalencyValidator parent, IEquivalencyAssertionOptions config)
+        {
+            var subject = context.Subject as DataColumn;
+            var expectation = context.Expectation as DataColumn;
+
+            if (expectation == null)
+            {
+                if (subject != null)
+                {
+                    return AssertionScope.Current.FailWith($"Expected value to be null, but found {subject}");
+                }
+            }
+            else
+            {
+                if (subject == null)
+                {
+                    return AssertionScope.Current.FailWith("Expected value to be non-null, but found null");
+                }
+
+                var dataSetConfig = config as DataEquivalencyAssertionOptions<DataSet>;
+                var dataTableConfig = config as DataEquivalencyAssertionOptions<DataTable>;
+                var dataColumnConfig = config as DataEquivalencyAssertionOptions<DataColumn>;
+
+                if (((dataSetConfig != null) && dataSetConfig.ShouldExcludeColumn(subject))
+                 || ((dataTableConfig != null) && dataTableConfig.ShouldExcludeColumn(subject))
+                 || ((dataColumnConfig != null) && dataColumnConfig.ShouldExcludeColumn(subject)))
+                {
+                    return true;
```

Comment:
🙃 I prefer to not jump out of a long method like this. [Single entry, single exit ](https://csharpcodingguidelines.com//maintainability-guidelines/#AV1540)is usually much more readable.

Reply:
Fair enough. I have restructured all of the methods using this pattern.